src/lib/components/composites/paper-components/paper-view/PaperNavigationButton.svelte

```diff
@@ -23,19 +23,23 @@
         direction,
         loadingProject,
         loadingProjectPaper,
-        loading = $bindable(false),
         paperQueue = $bindable([]),
         nextProjectPaper = $bindable(undefined),
         previousProjectPaper = $bindable(undefined),
     }: Props = $props();
     let buttonLeftDisabled: boolean = $state(true);
     let buttonRightDisabled: boolean = $state(true);
-    const showLoadingSpinner = $state({ value: false });
-    const tooltipText = direction === "left" ? "Previous Paper" : "Next Paper";
 
-    $effect(() => {
-        loading = showLoadingSpinner.value;
-    });
+    let buttonStillLoading: boolean = $state(true);
+
+    const isDisabled: boolean = $derived(
+        projectPaperLoading.isLoading ||
+            (direction === "right" ? buttonRightDisabled : buttonLeftDisabled) ||
+            buttonStillLoading,
+    );
+    let isLoading: boolean = $derived(projectPaperLoading.isLoading || buttonStillLoading);
```

Comment:
What about:

```suggestion
    const isLoading: boolean = $derived(projectPaperLoading.isLoading || buttonStillLoading);
    const isDisabled: boolean = $derived(
        isLoading || (direction === "right" ? buttonRightDisabled : buttonLeftDisabled),
    );
```

1. You can use `const` for `isLoading` since it's never reassigned.
2. You can reuse the `isLoading` in the evaluation of `isDisabled`, since it uses the same variables to compute the `isDisabled` state.

Reply:
Good catch. Looks good.